packages/spacecat-shared-utils/src/log-wrapper.js

```diff
@@ -11,45 +11,46 @@
  */
 
 /**
- * Wraps the provided function to enhance the logging functionality by including the
- * `jobId` in each log statement.
+ * A wrapper function that enhances logging by appending the `jobId` to all log statements
+ * within the provided context. This allows better traceability and correlation of logs
+ * associated with a specific job.
  *
- * @param {function} fn - The original function that will be wrapped.
- * @returns {function(object, object): Promise<Response>} A new function that, when invoked,
- *  will automatically include the `jobId` in all log statements within the context.
+ * @param {function} fn - The original function to be wrapped.
+ * @returns {function(object, object): Promise<Response>} A wrapped function that includes
+ *  `jobId` in all log statements if `context.contextualLog` is used.
  */
 export function logWrapper(fn) {
   return async (message, context) => {
     const { log } = context;
 
     if (log) {
-      if (!message || !message.jobId) {
-        log.debug('Missing jobId, hence it will not be included in log messages.');
-        context.contextualLog = log;
+      if (message && 'jobId' in message) {
+        const { jobId } = message;
 
-        return fn(message, context);
+        // Enhance the log object to include jobId in all log statements
+        context.contextualLog = {
+          info: (...args) => {
+            log.info({ jobId, ...args });
+          },
+          error: (...args) => {
+            log.error({ jobId, ...args });
+          },
+          debug: (...args) => {
+            log.debug({ jobId, ...args });
+          },
+          warn: (...args) => {
+            log.warn({ jobId, ...args });
+          },
+          trace: (...args) => {
+            log.trace({ jobId, ...args });
+          },
+        };
```

Comment:
I remembered where I had seen the `silly` level: https://github.com/adobe/helix-log/blob/main/src/log.js#L961

(not saying we need it in this impl; just didn't want you to think I was crazy!)

Reply:
Interesting. The repository appears to be the log forwarder, as indicated by the forwarding logic found here: [coralogix.js](https://github.com/adobe/helix-log/blob/main/src/coralogix.js). To align with this behaviour, I can add the fatal, verbose, and silly log levels.